src/Exception/ConcurrencyExceptionFactory.php

```diff
@@ -0,0 +1,19 @@
+<?php declare(strict_types = 1);
+
+namespace Prooph\EventStore\Pdo\Exception;
+
+use Prooph\EventStore\Exception\ConcurrencyException;
+
+class ConcurrencyExceptionFactory
```

Comment:
This can be a named constructor on the concurrency exception itself, no need for an additional class

Reply:
It can't. ConcurrencyException is in another package which should not be aware of something like PDO.

The only way I could use a named constructor here would be to extend that exception. I slightly prefer this solution though.

Reply:
Oh you're right. I'm okay with having a factory then.